aws/aws.go

```diff
@@ -137,6 +137,24 @@ func (a *AwsAccount) CreateAwsAccount(ctx context.Context, db models.XODB) error
 	return err
 }
 
+// UpdateAWSAccount updates an AWS account for a user. It does no error
+// checking: the caller should check themselves that the AWS account exists.
+// Only the Pretty will be updated.
+func (a *AwsAccount) UpdateAWSAccount(ctx context.Context, tx *sql.Tx) error {
+	logger := jsonlog.LoggerFromContextOrDefault(ctx)
+	dbAwsAccount, err := models.AwsAccountByID(tx, a.Id)
+	if err != nil {
+		logger.Error("Failed to get AWS account in database.", nil)
+	} else {
+		dbAwsAccount.Pretty = a.Pretty
+		err := dbAwsAccount.Update(tx)
```

Comment:
By using `:=` here you shadow the `err` defined at line 145, so at line 155 you would still return `nil`.

Reply:
Please write unit tests, you should have caught this with unit tests.

Reply:
`err` is shadowed only if the `err` defined at line 145 is `nil`. It's the expected behaviour.